pkg/cloud/providerconfig.go

```diff
@@ -187,6 +187,23 @@ func DefaultPricing() *CustomPricing {
 }
 
 func SetCustomPricingField(obj *CustomPricing, name string, value string) error {
+
+	// shared overhead costs go into SharedCosts with key "total" in CustomPricing
+	if name == "SharedOverhead" {
+
+		value = sanitizePolicy.Sanitize(value)
+		val := reflect.ValueOf(value)
+
+		if reflect.ValueOf(obj.SharedCosts["total"]).Type() != val.Type() {
+			return fmt.Errorf("cannot insert value into custom pricing shared costs")
+		}
+
+		obj.SharedCosts["total"] = value
+
+		return nil
+
+	}
```

Comment:
I don't like this at all, but I don't see any other place to put it.

Reply:
I just talked with Niko about this. A possible alternative is to actually add a `SharedOverhead` field to the `CustomPricing` struct and then check all uses of the `SharedCosts` map and replace with `SharedOverhead`. According to Niko, even though `SharedCosts` is a map it only ever has one thing in it (the shared overhead cost). You'll have to confirm that. If that is indeed the case then `SharedCosts` can be removed entirely and can be replaced with `SharedOverhead` which doesn't have to be a map.

Doing this would totally eliminate this nastiness but it does involve a more substantial code change (in other repos, too). I think it is a good idea for long term code health.

Does that make sense? What do you think?

Reply:
Yes, that makes complete sense. I'll confirm that there's no other elements being put in `SharedCosts` and, if that's the case, see about replacing it with `SharedOverhead`.